src/panels/config/integrations/integration-panels/thread/thread-config-panel.ts

```diff
@@ -467,10 +482,7 @@ export class ThreadConfigPanel extends SubscribeMixin(LitElement) {
     const network = (ev.currentTarget as any).network as ThreadNetwork;
     const router = (ev.currentTarget as any).router as ThreadRouter;
     const otbr = (ev.currentTarget as any).otbr as OTBRInfo;
-    const index =
-      network.dataset && router.border_agent_id
-        ? Number(ev.detail.index)
-        : Number(ev.detail.index) + 1;
+    const index = Number(ev.detail.index);
```

Comment:
@agners I think this change is the cause of Issue #22306 

`network.dataset` is not set there and the indexing on the resulting menu is off by 1

Reply:
Hm, we discussed this above, and from what I can tell every OTBR entry should actually have a dataset :thinking: 

https://github.com/home-assistant/frontend/pull/22183#discussion_r1784383134

Reply:
It would seem in this case that the router has not been assigned to any network yet. I suppose that is why its missing a dataset? 

[config_entry-thread-4f46c2ed6336e72b15a52386aa9ca930 (3).json](https://github.com/user-attachments/files/17327190/config_entry-thread-4f46c2ed6336e72b15a52386aa9ca930.3.json)

Reply:
Yeah so that is the thing: From what I understand how the OTBR integration works, it will either join the preferred network or create a new one. Looking at the code I don't understand how you can get into any other state :see_no_evil:  I must misunderstand something :thinking: 

Did you add that OTBR recently?

Reply:
I added a new device to the odroid.local OTBR today (well previously it should have been configured with a different device but been disconnected for a while and preferred network was already set as well).  Bringing up the new device did not add it to preferred network.

Reply:
I just can't reproduce this :see_no_evil:  For me, the OTBR integration provides a dataset.

So I've stopped an OTBR add-on running on a different system, deleted the integration and credentials on my test system. I then added the OTBR again. But this immediately imported the credentials.

The only way I can imagine that to fail is if somehow fetching the credentials or setting up a new Thread network fails. But this should lead to log entries in your Core when adding the OTBR integration :thinking:  Are the Core logs clean?

Are you using OTBR in a container? If so which OTBR POSIX hash is that built from? The behavior really should be the same since all patches are upstream.

In the end, I can try to fix the frontend by "feel", but I really would like to understand how things get there, and actually test my fix :see_no_evil:

Reply:
Hm, ok I see, we setup a new active dataset only when the config entry gets created. So if the OTBR integration is there, but the add-on/container cleared it's network config, then we just try to read the dataset, which won't return a valid dataset.

Reply:
In this case was using HA addon. The integration was previously configured and active, but addon had been offline for atleast a few weeks or more (I loose track as too many instances and test setups running).  Normally it keeps the dataset when swapping to new radio device so not sure what happened to the dataset in first place!

 I just deleted and reinstalled integration now and it added device to preferred network automatically. I have however in the past had to manually use the "add to preferred" .